src/python/turicreate/toolkits/_tf_utils.py

```diff
@@ -0,0 +1,105 @@
+# -*- coding: utf-8 -*-
+# Copyright © 2019 Apple Inc. All rights reserved.
+#
+# Use of this source code is governed by a BSD-3-clause license that can
+# be found in the LICENSE.txt file or at https://opensource.org/licenses/BSD-3-Clause
+
+from __future__ import print_function as _
+from __future__ import division as _
+from __future__ import absolute_import as _
+import numpy as np
+	
+def convert_shared_float_array_to_numpy(array):
+	"""
+	The initialization from C++ implementation is mapped to SharedFloatArray 
+	in Python through Pybind. It must be converted to numpy arrays to be used
+	in TensorFlow.
+
+	All the weights from the C++ implementation are four dimensional.
+
+	"""
+	return np.array(array, copy=False)
+
+def convert_conv1d_coreml_to_tf(conv_weights):
+	"""
+	The Convolutional weights in CoreML specification is sent as [C, H, 1, W] 
```

Comment:
`[C, H, 1, W]` doesn't look right to me. 

From the `NeuralNetwork.proto` spec:
```
    /**
     * Weights associated with this layer.
     * If convolution (``isDeconvolution == false``), weights have the shape
     * ``[outputChannels, kernelChannels, kernelHeight, kernelWidth]``, where kernelChannels == inputChannels / nGroups
     * If deconvolution (``isDeconvolution == true``) weights have the shape
     * ``[kernelChannels, outputChannels / nGroups, kernelHeight, kernelWidth]``, where kernelChannels == inputChannels
     */
```